docs/rfcs/0623-redis-backend.md

```diff
@@ -0,0 +1,237 @@
+- Proposal Name: "redis_backend"
+- Start Date: 2022-08-31
+- RFC PR: [datafuselabs/opendal#0623](https://github.com/datafuselabs/opendal/pull/0623)
+- Tracking Issue: [datafuselabs/opendal#0000](https://github.com/datafuselabs/opendal/issues/0000)
+
+# Summary
+
+Use [redis](https://redis.io) as a backend of OpenDAL.
+
+# Motivation
+
+Redis is a fast, in-memory cache with persistent and distributed storage functionalities. It's widely used in production.
+
+Users also demand more backend support. Redis is a good candidate.
+
+# Guide-level explanation
+
+Users only need to provide the network address, username and password to create a Redis Operator. Then everything will act as other operators do.
+
+```rust
+use opendal::services::redis::Builder;
+use opendal::Operator;
+
+let builder = Builder::default();
+
+// set the endpoint of redis server
+builder.endpoint("redis://127.0.0.1:2333");
+// set the username of redis
+builder.username("example");
+// set the password
+builder.password(&std::env::var("OPENDAL_REDIS_PASSWORD").expect("env OPENDAL_REDIS_PASSWORD not set"));
+// root path
+builder.root("/example/");
+
+let rds: opendal::services::redis::RedisAccessor = builder.build().unwrap();
+let op = Operator::new(rds);
+
+// congratulations, you can use `op` just like any other operators!
+```
+
+# Reference-level explanation
+
+To ease the development, [redis-rs](https://crates.io/crates/redis) will be used.
+
+Redis offers a key-value view, so the path of files could be represented as the key of the key-value pair.
+
+Files in opendal will be represented as `Hash` in Redis, a kind of semi-structured data type:
+
+```
+                                           ┌───────────────────────────┐
+                                           │  "mode" : "file"          │
+Key: /home/monika/                         ├───────────────────────────┤
+                                           │  "last_modified" :        │
+Key: /home/monika/poem0.txt ─────────────► │  "<rfc3339 timestamp>"    │
```

Comment:
I'm not sure if it's better to use `bincode` to sterilize our metadata directly. Maybe we can have a bench for them?